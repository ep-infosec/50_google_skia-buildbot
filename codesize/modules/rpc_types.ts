// DO NOT EDIT. This file is automatically generated.

export interface BinaryRPCRequest {
	binary_name: string;
	compile_task_name: string;
	commit: string;
	patch_issue: string;
	patch_set: string;
}

export interface BloatyOutputMetadata {
	version: number;
	timestamp: string;
	swarming_task_id: string;
	swarming_server: string;
	task_id: string;
	task_name: string;
	compile_task_name: string;
	compile_task_name_no_patch?: string;
	binary_name: string;
	bloaty_cipd_version: string;
	bloaty_args: string[] | null;
	bloaty_diff_args?: string[] | null;
	patch_issue: string;
	patch_server: string;
	patch_set: string;
	repo: string;
	revision: string;
	commit_timestamp: string;
	author: string;
	subject: string;
}

export interface TreeMapDataTableRow {
	name: string;
	parent: string;
	size: number;
}

export interface BinaryRPCResponse {
	metadata: BloatyOutputMetadata;
	rows: TreeMapDataTableRow[];
}

export interface BinarySizeDiffRPCRequest {
	binary_name: string;
	compile_task_name: string;
	commit: string;
	patch_issue: string;
	patch_set: string;
}

export interface BinarySizeDiffRPCResponse {
	metadata: BloatyOutputMetadata;
	raw_diff: string;
}

export interface Binary {
	metadata: BloatyOutputMetadata;
}

export interface BinariesFromCommitOrPatchset {
	binaries: Binary[];
	commit: string;
	patch_issue: string;
	patch_set: string;
}

export interface MostRecentBinariesRPCResponse {
	binaries: BinariesFromCommitOrPatchset[] | null;
}
