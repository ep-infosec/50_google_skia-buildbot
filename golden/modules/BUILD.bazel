load("//infra-sk:index.bzl", "karma_test", "sass_library", "ts_library")

karma_test(
    name = "common_test",
    src = "common_test.ts",
    deps = [
        ":common_ts_lib",
        "//golden/modules/search-controls-sk",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "common_ts_lib",
    srcs = ["common.ts"],
    visibility = ["//visibility:public"],
    deps = [
        ":rpc_types_ts_lib",
        "//golden/modules/search-controls-sk",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "demo_util_ts_lib",
    srcs = ["demo_util.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "rpc_types_ts_lib",
    srcs = ["rpc_types.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "settings_ts_lib",
    srcs = ["settings.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "shared_demo_data_ts_lib",
    srcs = ["shared_demo_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["@npm//common-sk"],
)

sass_library(
    name = "colors_sass_lib",
    srcs = ["colors.scss"],
    visibility = ["//visibility:public"],
)
