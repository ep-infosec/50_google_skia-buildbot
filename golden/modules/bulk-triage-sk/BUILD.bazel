load("//infra-sk:index.bzl", "karma_test", "sk_demo_page_server", "sk_element", "sk_element_puppeteer_test", "sk_page", "ts_library")

sk_element(
    name = "bulk-triage-sk",
    sass_deps = [
        "//golden/modules:colors_sass_lib",
        "@npm//:node_modules/elements-sk/colors.scss",
    ],
    sass_srcs = ["bulk-triage-sk.scss"],
    ts_deps = [
        "//golden/modules:common_ts_lib",
        "//golden/modules:rpc_types_ts_lib",
        "//infra-sk/modules/ElementSk:index_ts_lib",
        "@npm//elements-sk",
        "@npm//lit-html",
    ],
    ts_srcs = [
        "bulk-triage-sk.ts",
        "index.ts",
    ],
    visibility = ["//visibility:public"],
)

sk_page(
    name = "bulk-triage-sk-demo",
    html_file = "bulk-triage-sk-demo.html",
    sk_element_deps = [":bulk-triage-sk"],
    ts_deps = [
        ":test_data_ts_lib",
        "@npm//common-sk",
        "@npm//fetch-mock",
        "//golden/modules:rpc_types_ts_lib",
    ],
    ts_entry_point = "bulk-triage-sk-demo.ts",
)

ts_library(
    name = "bulk-triage-sk_po_ts_lib",
    srcs = ["bulk-triage-sk_po.ts"],
    visibility = ["//visibility:public"],
    deps = [
        "//infra-sk/modules/page_object:page_object_element_ts_lib",
        "//infra-sk/modules/page_object:page_object_ts_lib",
        "@npm//elements-sk",
    ],
)

sk_element_puppeteer_test(
    name = "bulk-triage-sk_puppeteer_test",
    src = "bulk-triage-sk_puppeteer_test.ts",
    sk_demo_page_server = ":demo_page_server",
    deps = [
        ":bulk-triage-sk_po_ts_lib",
        "//puppeteer-tests:util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//puppeteer",
    ],
)

karma_test(
    name = "bulk-triage-sk_test",
    src = "bulk-triage-sk_test.ts",
    deps = [
        ":bulk-triage-sk",
        ":bulk-triage-sk_po_ts_lib",
        ":test_data_ts_lib",
        "//golden/modules:rpc_types_ts_lib",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//fetch-mock",
    ],
)

sk_demo_page_server(
    name = "demo_page_server",
    sk_page = ":bulk-triage-sk-demo",
)

ts_library(
    name = "test_data_ts_lib",
    srcs = ["test_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["//golden/modules:rpc_types_ts_lib"],
)
