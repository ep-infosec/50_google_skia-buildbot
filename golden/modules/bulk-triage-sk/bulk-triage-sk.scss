@import 'node_modules/elements-sk/colors.scss';
@import '../colors';

bulk-triage-sk {

  button {
    &.selected {
      background-color: var(--dark-white);
      border: none;
      box-shadow: none;
    }

    &.positive {
      &.selected check-circle-icon-sk {
        fill: var(--triaged-positive);
      }
    }

    &.negative {
      &.selected cancel-icon-sk {
        fill: var(--triaged-negative);
      }
    }

    &.untriaged {
      &.selected help-icon-sk {
        fill: var(--untriaged);
      }
    }

    &.closest {
      &.selected view-agenda-icon-sk {
        fill: #1F78B4;
      }
    }
  }
}
