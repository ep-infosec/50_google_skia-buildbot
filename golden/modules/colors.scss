/* Common Gold Colors via CSS Variables.

 Can be imported via:

    @import '../colors';
*/

html {
  --gold: rgb(217, 95, 2);
  --bright-orange: rgb(255, 127, 0);

  --light-error-red: rgb(255, 230, 230);
  --transparent-white: rgba(255, 255, 255, 0);

  --triaged-positive: #1B9E77;
  --triaged-negative: #E7298A;
  --untriaged: #A6761D;
}
