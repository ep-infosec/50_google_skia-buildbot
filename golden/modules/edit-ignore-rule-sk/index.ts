import './edit-ignore-rule-sk';
