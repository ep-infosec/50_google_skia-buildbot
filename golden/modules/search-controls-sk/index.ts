import './search-controls-sk';

export { SearchCriteriaToHintableObject, SearchCriteriaFromHintableObject } from './search-controls-sk';
