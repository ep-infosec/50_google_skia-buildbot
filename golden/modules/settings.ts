/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
