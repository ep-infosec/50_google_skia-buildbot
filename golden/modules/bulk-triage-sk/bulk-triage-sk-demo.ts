import './index';
import { $$ } from 'common-sk/modules/dom';
import fetchMock from 'fetch-mock';
import { deepCopy } from 'common-sk/modules/object';
import { bulkTriageDeltaInfos } from './test_data';
import { BulkTriageSk } from './bulk-triage-sk';
import { TriageResponse } from '../rpc_types';

const handleTriaged = () => {
  const log = $$<HTMLPreElement>('#event_log')!;
  log.textContent += 'Did triage.\n';
};

const handleCancelled = () => {
  const log = $$<HTMLPreElement>('#event_log')!;
  log.textContent += 'Cancelled.\n';
};

const ele = new BulkTriageSk();
ele.bulkTriageDeltaInfos = deepCopy(bulkTriageDeltaInfos);
ele.addEventListener('bulk_triage_invoked', handleTriaged);
ele.addEventListener('bulk_triage_cancelled', handleCancelled);
$$('#default')!.appendChild(ele);

const eleCL = new BulkTriageSk();
eleCL.bulkTriageDeltaInfos = deepCopy(bulkTriageDeltaInfos);
eleCL.changeListID = '1234567';
eleCL.crs = 'github';
eleCL.addEventListener('bulk_triage_invoked', handleTriaged);
eleCL.addEventListener('bulk_triage_cancelled', handleCancelled);
$$('#changelist')!.appendChild(eleCL);

const response: TriageResponse = { status: 'ok' };
fetchMock.post({ url: '/json/v3/triage' }, {
  status: 200,
  body: response,
});
