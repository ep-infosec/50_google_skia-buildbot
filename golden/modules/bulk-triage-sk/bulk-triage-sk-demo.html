<!DOCTYPE html>
<html>
  <head>
    <title>bulk-triage-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  </head>
  <body class="body-sk">
    <h1>bulk-triage-sk</h1>
    <div id="default"></div>

    <h1>bulk-triage-sk for CL</h1>
    <div id="changelist"></div>

    <h2>Events</h2>
    <pre id="event_log"></pre>
  </body>
</html>
